### PR TITLE
Improve non-tagged network ACL support and fix tagging

### PR DESCRIPTION
**Pull request by willthames**
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
ec2_vpc_nacl

##### ANSIBLE VERSION
```
ansible 2.3.0 (devel 6ec0369c26) last updated 2017/01/11 13:34:13 (GMT +1000)
  config file = 
  configured module search path = Default w/o overrides
```

##### SUMMARY
* Allow network ACLs to be manipulated by ID
* Fix tag handling so that it doesn't assume that there are always tags
* Move tag manipulation to common code suitable for more resources
* Clean up some flake8 issues

### CODE REVIEW

**Review by mmochan — APPROVED** [review]
shipit

File: lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py [hunk 1]
```diff
@@ -13,6 +13,16 @@
...
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.ec2 import boto3_tag_list_to_ansible_dict, HAS_BOTO3, update_aws_tags
+
+try:
+    import botocore
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
```
Comment (mattclay) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py]: CI failure due to new import tests added today:

```
2017-02-02 23:38:07 ============================================================================
2017-02-02 23:38:07 lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py
2017-02-02 23:38:07 ============================================================================
2017-02-02 23:38:07 ERROR: Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 17
2017-02-02 23:38:07 ERROR: Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 18
2017-02-02 23:38:07 ERROR: Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 19
2017-02-02 23:38:07 ERROR: Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 21
```

**Comment by mattclay** [comment]
Closing and re-opening to trigger CI.

**Comment by mattclay** [comment]
CI failed for the aforementioned module validator errors (import placement).

**Comment by mattclay** [comment]
CI failure due to module docs validation errors:

```
2017-02-22 04:32:02 ============================================================================
2017-02-22 04:32:02 lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py
2017-02-22 04:32:02 ============================================================================
2017-02-22 04:32:02 ERROR:309:version_added for new option (nacl_id) should be 2.3. Currently 0.0
2017-02-22 04:32:02 ERROR:106:Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 17
2017-02-22 04:32:02 ERROR:106:Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 18
2017-02-22 04:32:02 ERROR:106:Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 19
2017-02-22 04:32:02 ERROR:106:Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 21
```

**Comment by willthames** [comment]
Superseded by #23112

**Pull request closed without merging**